File: EEDU-Backend/src/main/java/de/gaz/eedu/course/appointment/entry/model/AssignmentModel.java
package de.gaz.eedu.course.appointment.entry.model;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.time.Instant;
import java.util.Objects;

public record AssignmentModel(@NotNull String description, @Nullable Long publish, @NotNull Long submitUntil)
{

    public AssignmentModel(@NotNull String description, @Nullable Instant publish, @NotNull Instant submitUntil)
    {
        this(description, Objects.isNull(publish) ? null : publish.toEpochMilli(), submitUntil.toEpochMilli());
    }
}

File: EEDU-Frontend/src/app/user/courses/appointment/entry/assignment-model.ts
export interface GenericAssignment {
    description: string;
    publish: number;
    submitUntil: number;
}

export class AssignmentModel {

    constructor(
        private _description: string,
        private _publish: number,
        private _submitUntil: number
    ) {}

    public get description(): string {
        return this._description;
    }

    public get publish(): Date {
        return new Date(this._publish);
    }

    public get submitUntil(): Date {
        return new Date(this._submitUntil);
    }

    public static fromObject(obj: GenericAssignment): AssignmentModel {
        return new AssignmentModel(obj.description, obj.publish, obj.submitUntil);
    }
}
